### ISSUE TITLE
[Feature] Domain name support in endpoint field for WireGuard protocol in Clash Premium

### ISSUE THREAD

**Issue by ricky9w** [Dreamacro/clash#2551]
### Verify steps

- [X]  我已经在 [Issue Tracker](……/) 中找过我要提出的请求 I have searched on the [issue tracker](……/) for a related feature request. 
- [X]  我已经仔细看过 [Documentation](https://github.com/Dreamacro/clash/wiki/) 并无法自行解决问题 I have read the [documentation](https://github.com/Dreamacro/clash/wiki/) and was unable to solve the issue. 

### Description

I would like to express my gratitude for the recent addition of WireGuard protocol support to Clash Premium. It's great to see that the product is constantly improving and evolving.

However, I have encountered an issue where the current implementation does not allow using domain names in the "endpoint" field. When attempting to use a domain name as the endpoint, the following error occurs:

```
[WARN] [TCP] dial failed error=IPC error -22: failed to set endpoint mydomain:12345: couldn't find ip: mydomain
```

I believe it would be very convenient if domain name support was added to the "endpoint" field, especially for scenarios that use DDNS services.

Additionally, if domain name support is added, would it be possible to include a feature that dynamically resolves and updates the endpoint IP address for the above domain name?

Thank you for your attention to this matter, and I look forward to your response.

Best regards,
Richard

**Comment by Dreamacro** [comment]
It looks like the dns resolve failed, is the domain with only ipv6?

**Comment by ricky9w** [comment]
> It looks like the dns resolve failed, is the domain with only ipv6?

Oh yes, this is an IPv6-only domain name. It's becoming increasingly difficult for home broadband users to obtain public IPv4 addresses. I apologize for not explaining this clearly in the problem description.

**Comment by Dreamacro** [comment]
Open the two ipv6 fields in the clash configuration https://github.com/Dreamacro/clash/wiki/Configuration

**Comment by ricky9w** [comment]
> Open the two ipv6 fields in the clash configuration https://github.com/Dreamacro/clash/wiki/Configuration

I left two settings related to IPv6 blank, thinking that they were enabled by default. The issue was resolved after I manually set them to "true". Thank you very much.

By the way, does the current configuration support periodic re-resolution of the endpoint domain name and update of its IP address, just like what this tool does:
https://github.com/WireGuard/wireguard-tools/tree/96e42feb3f41e2161141d4958e2637d9dee6f90a/contrib/reresolve-dns
